### PR TITLE
Add AddressSanitizer support.

### PR DESCRIPTION
**Pull request by steven-johnson**
Like the previous MSAN and TSAN support. Not quite ready for review yet; opening a PR for comments and experimentation.

### CODE REVIEW

File: src/CodeGen_LLVM.cpp [hunk 1]
```diff
@@ -1071,12 +1071,24 @@ void CodeGen_LLVM::optimize_module() {
     }
 #endif
 
+    if (get_target().has_feature(Target::ASAN)) {
+        auto addAddressSanitizerPass = [](const PassManagerBuilder &builder, legacy::PassManagerBase &pm) {
+            // TODO: we we (also/instead) want createAddressSanitizerModulePass?
+            pm.add(createAddressSanitizerFunctionPass());
```
Comment (alinas) [inline src/CodeGen_LLVM.cpp]: AFAIU asan initialization is done in createAddressSanitizerModulePass, so yes.

Comment (steven-johnson) [reply src/CodeGen_LLVM.cpp]: Is that yes (both) or yes (instead)?

Comment (alinas) [reply src/CodeGen_LLVM.cpp]: Yes both.

**Comment by steven-johnson** [comment]
> DebugIR pass 

Yes yes yes, that would be excellent. Please keep us up to date :-)

**Comment by steven-johnson** [comment]
I think this is close enough to land. PTAL.

**Comment by steven-johnson** [comment]
Will land as soon as things are green.

**Pull request merged**